camel/toolkits/notion_toolkit.py

```diff
@@ -0,0 +1,279 @@
+            "type", "Unknown Type"
+        )
+
+    if len(block.get(block.get("type"), {}).get("caption", [])) > 0:
+        caption = get_plain_text_from_rich_text(
+            block[block["type"]]["caption"]
+        )
+        return caption + ": " + source
+
+    return source
+
+
+class NotionToolkit(BaseToolkit):
+    r"""A toolkit for retrieving information from the user's notion pages.
+
+    Attributes:
+        notion_token (Optional[str], optional): The notion_token used to
+            interact with notion APIs.(default: :obj:`None`)
+        notion_client (module): The notion module for interacting with
+            the notion APIs.
+    """
+
+    def __init__(
+        self,
+        notion_token: Optional[str] = None,
+    ) -> None:
+        r"""Initializes the NotionToolkit.
+
+        Args:
+            notion_token (Optional[str], optional): The optional notion_token
+            used to interact with notion APIs.(default: :obj:`None`)
+        """
+        from notion_client import Client
+
+        self.notion_token = notion_token or os.environ.get("NOTION_TOKEN")
+
+        self.notion_client = Client(auth=self.notion_token)
+
+    def list_all_users(self) -> List[dict]:
+        r"""Lists all users via the Notion integration.
+
+        Returns:
+            A list of user objects with type, name, and workspace.
+        """
+        all_users_info: List[dict] = []
+        cursor = None
+
+        while True:
+            response = cast(
+                dict,
+                self.notion_client.users.list(start_cursor=cursor),
+            )
+            all_users_info.extend(response["results"])
+
+            if not response["has_more"]:
+                break
+
+            cursor = response["next_cursor"]
+
+        final_list = [
+            {
+                "type": user["type"],
+                "name": user["name"],
+                "workspace": user.get(user.get("type"), {}).get(
+                    "workspace_name", ""
+                ),
+            }
+            for user in all_users_info
+        ]
+
+        return final_list
+
+    def list_all_pages(self) -> List[dict]:
+        r"""Lists all pages in the Notion workspace.
+
+        Returns:
+            A list of page objects with title and id.
+        """
+        all_pages_info: List[dict] = []
+        cursor = None
+
+        while True:
+            response = cast(
+                dict,
+                self.notion_client.search(
+                    filter={"property": "object", "value": "page"},
+                    start_cursor=cursor,
+                ),
+            )
+            all_pages_info.extend(response["results"])
+
+            if not response["has_more"]:
+                break
+
+            cursor = response["next_cursor"]
+
+        final_list = [
+            {
+                "id": page["id"],
+                "title": next(
+                    (
+                        title["text"]["content"]
```

Comment:
use `.get` would be more safe?

Reply:
All the "if" conditions use get to make it safe, and when it falls in any "if" condition, then the fields must be there.